bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/common/WrappedScheduledExecutorService.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2014,2017 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.common;
```

Comment:
Do we need/want to publicly expose this or could it be moved to an internal package?

Reply:
@Hilbrand Did you see this question?

Reply:
I kind of assumed his was a question to the wider smarthome audience ;-)
I put it here as it might be useful in other places, although I have no specific use cases in mind. The class is already package private to limit visibility. Other options might be to make it an internal class ThreadPoolManager, but that would limit future usages outside that class. But I don't have a particular opinion in this case as my knowledge of the overall system is limited.

Reply:
> I kind of assumed his was a question

No, it was a hidden order 😛 

> The class is already package private to limit visibility.

Moving it to an unexported package is a much stronger means of OSGi to restrict visibility. I'd therefore suggest to move it to ...internal.common.

If we one day have a request for reuse, we can still consider making it part of the API. Until then, it is always safer to keep the API as small as possible :-)

Reply:
Ok order executed :smile:

(That's what I mean by not know the whole system. I completely overlooked the internal packages, and only looked in the common package itself, and not one place higher)